tests/test_vault.py

```diff
@@ -0,0 +1,145 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"vault secrets tests"
+
+import os
+import unittest
+import tempfile
+import base64
+
+import kapitan.cached as cached
+from kapitan.refs.base import RefController, RefParams, Revealer
+from kapitan.refs.secrets.vault import  vault_obj, VaultSecret
+
+import hvac
+
+
+def initialize():
+    """
+    Initiaize Vault and return root token
+    """
+    client = hvac.Client()
+    init = client.sys.initialize()
+    client.sys.submit_unseal_keys(init['keys'])
+    return init['root_token']
+
+ROOT_TOKEN= initialize()
+print(ROOT_TOKEN)
+client = hvac.Client(token = ROOT_TOKEN)
+client.sys.enable_secrets_engine(backend_type='kv-v2',path='secret')
+test_policy = '''
+path "secret/*" {
+  capabilities = ["read", "list"]
+}
+'''
+policy = 'test_policy'
+client.sys.create_or_update_policy(name=policy,policy=test_policy)
+USERNAME = 'test_user'
+PASSWORD = 'test_password'
+client.sys.enable_auth_method('userpass')
+client.create_userpass(username=USERNAME, password=PASSWORD, policies=[ policy ])
+client.sys.enable_auth_method('approle')
+client.create_role('test_role')
+ROLE_ID = client.get_role_id('test_role')
+SECRET_ID = client.create_role_secret_id('test_role')['data']['secret_id']
+TOKEN = client.create_token(policies=[policy],lease='1h')['auth']['client_token']
+
+REFS_HOME = tempfile.mkdtemp()
+REF_CONTROLLER = RefController(REFS_HOME)
+REVEALER = Revealer(REF_CONTROLLER)
+
+
+class VaultSecretTest(unittest.TestCase):
+    "Test Vault Secret"
+
+    def test_token_authentication(self):
+        '''
+        Authenticate using token
+        '''
+        os.environ['VAULT_TOKEN'] = TOKEN
+        test_client = vault_obj()
+        self.assertTrue(test_client.is_authenticated())
+        test_client.adapter.close()
+
+    def test_userpss_authentication(self):
+        '''
+        Authenticate using userpass
+        '''
+        os.environ['VAULT_AUTHTYPE'] = 'userpass'
+        os.environ['VAULT_USER'] = USERNAME
+        os.environ['VAULT_PASSWORD'] = PASSWORD
+        test_client = vault_obj()
+        self.assertTrue(test_client.is_authenticated())
+        test_client.adapter.close()
+
+    def test_approle_authentication(self):
+        '''
+        Authenticate using approle
+        '''
+        os.environ['VAULT_AUTHTYPE'] = 'approle'
+        os.environ['VAULT_ROLE_ID'] = ROLE_ID
+        os.environ['VAULT_SECRET_ID'] = SECRET_ID
+        test_client = vault_obj()
+        self.assertTrue(test_client.is_authenticated())
+        test_client.adapter.close()
+
+    def test_vault_write_reveal(self):
+        '''
+        Write secret, confirm secret file exists, reveal and compare content
```

Comment:
it defeats the purpose of unit testing. you testing multiple activities in one test function. 

How exactly are you testing the following methods? 

- reveal(self)
- decrypt
- dump

Reply:
We need tests for the cli option as well. test_cli.py needs those test cases.

Reply:
For the first iteration the tests are good. please resolve this.